### ISSUE TITLE
Add ConfigurableFileNesting capability to managed projects

### ISSUE THREAD

**Issue by swesonga** [dotnet/project-system#6027]
The file nesting implementation will be enabled only on projects with the ConfigurableFileNesting project capability. This capability needs to be defined for the managed project system to light up the file nesting feature.

This change would be merged only after the file nesting feature branch is merged into CPS master.
